src/index.js

```diff
@@ -53,57 +69,73 @@ export default source => {
+            if (expression.type === 'MemberAccess') 
+              return expression.memberName
+        }
+      })
+
+    //CONSTRUCTOR
+    if (!node.name) {
+      handleSpecials(node)
+    }
+
     return {
       name: graphNodeName(node.name),
       callees:functionCallees,
-      send: functionCallees.some(callee => {
-        return (callee.name || callee.property && callee.property.name) === 'send'
-      }),
-      transfer: functionCallees.some(callee => {
-        return (callee.name || callee.property && callee.property.name) === 'transfer'
-      }),
-      constant: node.modifiers && node.modifiers.some(propEquals('name', 'constant')),
-      internal: node.modifiers && node.modifiers.some(propEquals('name', 'internal')),
-      view: node.modifiers && node.modifiers.some(propEquals('name', 'view')),
-      pure: node.modifiers && node.modifiers.some(propEquals('name', 'pure')),
-      payable: node.modifiers && node.modifiers.some(propEquals('name', 'payable'))
+      send: functionCallees.includes('send'),
+      transfer: functionCallees.includes('transfer'),
+      constant: node.stateMutability && node.stateMutability === 'constant',
+      internal: node.visibility && node.visibility === 'internal',
+      view: node.stateMutability && node.stateMutability === 'view',
+      pure: node.stateMutability && node.stateMutability === 'pure',
+      payable: node.stateMutability && node.stateMutability === 'payable',
+      event: node.type && node.type === 'EventDefinition'
     }
   })
 
-  // console.log(JSON.stringify(ast, null, 2))
-  // console.log(JSON.stringify(analyzedNodes, null, 2))
-
   // generate a graph
   var digraph = new Graph()
-  analyzedNodes.forEach(({ name, callees, send, constant, internal, view, pure, transfer, payable }) => {
+  analyzedNodes.forEach(({ name, callees, send, constant, internal, view, pure, transfer, payable, event }) => {
 
     // node
     digraph.setNode(graphNodeName(name),
-      send ? { color: COLORS.SEND } :
+      event ? EVENT_NODE_STYLE :
+      send ? { color: COLORS.DEPRECATED } :
       constant ? { color: COLORS.CONSTANT } :
       internal ? { color: COLORS.INTERNAL } :
       view ? { color: COLORS.VIEW } :
       pure ? { color: COLORS.PURE } :
-      transfer ? { color: COLORS.TRANSFER } :
+      transfer ? { color: COLORS.DEPRECATED } :
       payable ? { color: COLORS.PAYABLE } :
       {}
     )
 
     // edge
     callees.forEach(callee => {
-      const calleeName = callee.property && callee.property.name || callee.name
-      digraph.setEdge(name, graphNodeName(calleeName))
+      digraph.setEdge(name, graphNodeName(callee))
     })
   })
 
   // add send node
   if(analyzedNodes.some(prop('send'))) {
-    digraph.setNode(SEND_NODE_NAME, SEND_NODE_STYLE)
+    digraph.setNode(DEPRECATED_NODE_NAMES['send'], DEPRECATED_NODE_STYLE)
+  }
+  if(analyzedNodes.some(prop('transfer'))) {
+    digraph.setNode(DEPRECATED_NODE_NAMES['transfer'], DEPRECATED_NODE_STYLE)
   }
```

Comment:
The `send` and `transfer` methods are already added to the graph above. Rather than overriding them, use conditionals to render them as desired the first time.

Reply:
These are not overrides, `send`  and `transfer` are not being added, as they are solidity account object native methods and not functions nor events defined in the contract. 
I am actually going to optimise it by iterating over the `DEPRECATED` array.

```
  DEPRECATED.forEach(name => {
    if(analyzedNodes.some(prop(name))) {
      digraph.setNode(deprecatedNodeName(name), DEPRECATED_NODE_STYLE)
    }
  })
```

Reply:
Indeed, they are however added indirectly by `digraph.setEdge` as they are native methods of solidity `address` object and are not defined in the analysed file nor it's ancestors.
For the record this is not a behaviour introduced by my changes, it's been like that before.
I'd argue to leave the overrides for now, as changing it would require significant changes to the application logic.

Reply:
Okay, that's reasonable.